### PR TITLE
refactor(core): move utxo recover method to separate file

### PR DESCRIPTION
**Pull request by OttoAllmendinger**
The class method implementation was using deprecated coroutine wrappers and
was way too long.

Issue: BG-34381

5e89226e (Otto Allmendinger, 36 minutes ago)
chore(core): apply prettier to abstractUtxoCoin

Issue: BG-34381

### CODE REVIEW

File: modules/core/src/v2/coins/abstractUtxoCoin.ts [hunk 1]
```diff
@@ -1,7 +1,7 @@
 /**
  * @prettier
  */
-import { Codes, VirtualSizes } from '@bitgo/unspents';
+import { Codes } from '@bitgo/unspents';
 import { UnspentType } from '@bitgo/unspents/dist/codes';
```
Comment (tylerlevine) [inline modules/core/src/v2/coins/abstractUtxoCoin.ts:5]: Hm, do we need to pull this out of `dist/codes`? Is it not exported? (Optional nit since this code is unchanged)

File: modules/core/src/v2/coins/abstractUtxoCoin.ts [hunk 2]
```diff
@@ -1788,8 +1778,8 @@ export abstract class AbstractUtxoCoin extends BaseCoin {
     return response;
   }
 
-  protected abstract getAddressInfoFromExplorer(address: string, apiKey?: string): Bluebird<AddressInfo>;
-  protected abstract getUnspentInfoFromExplorer(address: string, apiKey?: string): Bluebird<UnspentInfo[]>;
+  public abstract getAddressInfoFromExplorer(address: string, apiKey?: string): Bluebird<AddressInfo>;
+  public abstract getUnspentInfoFromExplorer(address: string, apiKey?: string): Bluebird<UnspentInfo[]>;
```
Comment (tylerlevine) [inline modules/core/src/v2/coins/abstractUtxoCoin.ts:1782]: I'm not sure we want to add this to the public api, since it's really supposed to only be used for recoveries. Instead, maybe keep these `protected` and pass the functions into `recover`?

File: modules/core/src/v2/coins/utxo/recover.ts [hunk 3]
```diff
@@ -0,0 +1,284 @@
...
+  coin: AbstractUtxoCoin,
+  params: RecoverParams,
+  keyArray: bitcoin.HDNode[],
+  basePath: string,
+  addressesById
+) {
+  const MAX_SEQUENTIAL_ADDRESSES_WITHOUT_TXS = params.scan || 20;
+  let numSequentialAddressesWithoutTxs = 0;
+
+  async function gatherUnspents(addrIndex: number) {
```
Comment (tylerlevine) [inline modules/core/src/v2/coins/utxo/recover.ts:37]: I'd probably refactor this to be iterative instead of recursive but let's leave that for another PR

File: modules/core/src/v2/coins/utxo/recover.ts [hunk 4]
```diff
@@ -0,0 +1,284 @@
...
+      const externalKey = coin.deriveKeys(baseKeyPath, externalChainCode);
+      const internalKey = coin.deriveKeys(baseKeyPath, internalChainCode);
+      queries.push(queryBlockchainUnspentsPath(coin, params, externalKey, '/0/0/' + externalChainCode, addressesById));
+      queries.push(queryBlockchainUnspentsPath(coin, params, internalKey, '/0/0/' + internalChainCode, addressesById));
+    }
+  });
+
+  // Execute the queries and gather the unspents
+  const queryResponses = await Promise.all(queries);
+  const unspents: any[] = _.flatten(queryResponses); // this flattens the array (turns an array of arrays into just one array)
```
Comment (tylerlevine) [inline modules/core/src/v2/coins/utxo/recover.ts:188]: `// this comment repeats something from the documentation`

😂

**Review by tylerlevine — APPROVED** [review]
couple small comments, I'm mostly interested in your thoughts on the addition to the public API for the explorer functions

File: modules/core/src/v2/coins/abstractUtxoCoin.ts (hunk 2 above)
Comment (OttoAllmendinger) [reply modules/core/src/v2/coins/abstractUtxoCoin.ts:1782]: if it really is only used for recoveries IMO we should move it to `/recover.ts` and use `switch` instead of subclassing

**Comment by OttoAllmendinger** [comment]
let's do further cleanups in separate PRs
